File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        for(String s : strs){
            char[] characters = s.toCharArray();
            Arrays.sort(characters);
            String key = new String(characters);
            if(!map.containsKey(key)){
                map.put(key, new ArrayList<>());
            }
            List<String> list = map.get(key);
            list.add(s);
        }
        return new ArrayList<>(map.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashMap = {}
        for word in strs:
            key = "".join(sorted(word))
            if key not in hashMap:
                hashMap[key] = []
            hashMap[key].append(word)
        return list(hashMap.values())